framer/InputLayer.coffee

```diff
@@ -0,0 +1,148 @@
+Utils = require "./Utils"
+
+{Layer} = require "./Layer"
+{TextLayer} = require "./TextLayer"
+{Events} = require "./Events"
+
+Events.EnterKey  = "EnterKey"
+Events.BackspaceKey  = "BackspaceKey"
+Events.InputValueChange = "InputValueChange"
+Events.InputFocus = "InputFocus"
+Events.InputBlur = "InputBlur"
+
+class exports.InputLayer extends TextLayer
+
+	constructor: (options={}) ->
+
+		_.defaults options,
+			backgroundColor: "#FFF"
+			width: 500
+			height: 100
+
+		super options
+
+		# Layer containing input element
+		@input = new Layer
+			backgroundColor: "transparent"
+			x: 30
+			width: @width - 60
+			height: @height
+			parent: @
+			name: "input"
+
+		if @multiLine
+			@_inputElement = document.createElement("textarea")
+			@input.y = 25
+		else
+			@_inputElement = document.createElement("input")
+
+		# The id serves to differentiate multiple input elements from one another.
+		# To allow styling the placeholder colors of seperate elements.
+		@_id = Utils.round(Utils.randomNumber(0, 100))
+		@_inputElement.className = "input" + @_id
+
+		@on "change:width", =>
+			@_inputElement.style.width = "#{@input.width}px}"
+
+		# Append element
+		@input._element.appendChild(@_inputElement)
+
+		# Match TextLayer defaults and type properties
+		@_setTextProperties()
+
+		# All inherited properties
+		textProperties =
+			{@fontFamily, @fontSize, @lineHeight, @fontWeight, @color, @backgroundColor, @width, @height}
+
+		for property, value of textProperties
+
+			@on "change:#{property}", (value) =>
+				@_setTextProperties()
+				@_setPlaceholder()
```

Comment:
See if you can achieve something similar by overriding the `@_setStyle` function of `TextLayer`